src/handlers/pokemonHandler.ts

```diff
@@ -52,9 +52,28 @@ const findPokemons = async (query: {
   return await pokemonModel.searchPokemons(whereClause, orderBy, skip, take);
 };
 
+const getPokemonTypesCount = async (userSub: string) => {
+  const result = await pokemonModel.getPokemonTypes(userSub);
+
+  const typesCount: { [key: string]: number } = {};
+
+  result.forEach((userPokemon) => {
+    userPokemon.Pokemon.profile?.types.forEach((type) => {
+      if (typesCount[type]) {
+        typesCount[type]++;
+      } else {
+        typesCount[type] = 1;
+      }
+    });
+  });
```

Comment:
This screams `reduce` to me, try using reduce to gain the same functionality - it is a great method that would help you a lot in situations like this.